desktop/libs/indexer/src/indexer/api3_tests.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Licensed to Cloudera, Inc. under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  Cloudera, Inc. licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tempfile
+import uuid
+import json
+from desktop.lib.paths import get_desktop_root
+from django.http import HttpRequest, QueryDict
+from indexer.api3 import guess_field_types
+from nose.tools import assert_equal, assert_true
+
+
+def test_local_file_header_reformatting_with_header():
+    username = 'test_user'
+    filename = "%s_%s:testbook1.csv;" % (username, uuid.uuid4())
+    temp_file = tempfile.NamedTemporaryFile(suffix='.csv', delete=False)
```

Comment:
Could you use StringIO instead? (even cleaner interface this way)

e.g. https://github.com/cloudera/hue/blob/master/desktop/core/src/desktop/lib/export_csvxls_tests.py#L93